packages/wdio-sync/src/index.js

```diff
@@ -83,23 +84,23 @@ const executeAsync = function (fn, repeatTest = 0, args = []) {
     }
 
     /**
-     * if we don't retry just return result
+     * handle promise response
      */
-    if (repeatTest === 0 || !result || typeof result.catch !== 'function') {
-        return new Promise(resolve => resolve(result))
+    if (result && typeof result.catch === 'function') {
```

Comment:
let's change to `if (result > 0 && typeof result.catch === 'function') {`

Reply:
Not sure I follow sorry. `result` here is the value returned from the hook so is probably not a number? We check it exists to prevent `typeof result.catch` throwing in error if `result` is `undefined`, `null` etc. Could you clarify?

Reply:
Damn my fault, though about repeatTest